Update canary upgrade docs (#10086)
istio/istio.io: content/en/docs/setup/upgrade/canary/index.md

---

Upgrading Istio can be done by first running a canary deployment of the new control plane, allowing you
to monitor the effect of the upgrade with a small percentage of the workloads before migrating all of the
traffic to the new version. This is much safer than doing an

> [in place upgrade](/docs/setup/upgrade/in-place/) and is the recommended upgrade method.

Reviewer: ```suggestion
[in-place upgrade](/docs/setup/upgrade/in-place/) and is the recommended upgrade method.
```

Author: Fixed.

  To get started, check out https://istio.io/latest/docs/setup/getting-started/
{{< /text >}}

> Note that when using revision-based upgrades jumping across two patch versions is supported (e.g. upgrading directly from
> version `1.8` to `1.10`). This is in contrast to in-place upgrades where it is required to upgrade to each intermediate patch
> release.
> 

Reviewer: ```suggestion
{{< idea >}}
When using revision-based upgrades, jumping across two patch versions is supported (e.g. upgrading directly from
version `1.9` to `1.11`). This is in contrast to in-place upgrades, where it is required to upgrade to each intermediate patch
release.
{{< /idea >}}

```

Author: Added.

The output confirms that the pod is using `istiod-canary` revision of the control plane.

> ## Stable revision labels

Reviewer 2: Should you also change the feature status page (add it to the list as well)?

Author: I added "experimental" back to the heading pending TOC graduation of this feature to Alpha, and added the feature to `data/features.yaml`.

## Stable revision labels (experimental)

Manually relabeling namespaces when moving them to a new revision can be tedious and error-prone.

> [Revision tags](/docs/reference/commands/istioctl/#istioctl-experimental-tag) solve this problem.
> [Revision tags](/docs/reference/commands/istioctl/#istioctl-experimental-tag) are stable identifiers that point to revisions and can be used to avoid relabeling namespaces. Rather than relabeling the namespace, a mesh operator can simply change the tag to point to a new revision. All namespaces labeled with that tag will be updated at the same time.

Reviewer: ```suggestion
[Revision tags](/docs/reference/commands/istioctl/#istioctl-experimental-tag) solve this problem. Revision tags are stable identifiers that point to revisions and can be used to avoid relabeling namespaces. Rather than relabeling the namespace, a mesh operator can simply change the tag to point to a new revision. All namespaces labeled with that tag will be updated at the same time.
```

They're graduating from Experimental in 1.11 yes?  Link may need updating too

Reviewer: Link is definitely wrong, it's currently `https://istio.io/latest/docs/reference/commands/istioctl/#istioctl-experimental-revision-tag`

Author: Good catch! Fixed the link.

    >}}

The cluster operator can view this mapping in addition to tagged namespaces through the `istioctl tag list` command:

{{< text bash >}}
$ istioctl tag list
TAG         REVISION NAMESPACES
prod-canary 1-10-0   ...
prod-stable 1-9-5    ...
{{< /text >}}

After the cluster operator is satisfied with the stability of the control plane tagged with `prod-canary`, namespaces labeled
`istio.io/rev=prod-stable` can be updated with one action by modifying the `prod-stable` revision tag to point to the newer
`1-10-0` revision.

{{< text bash >}}
$ istioctl tag set prod-stable --revision 1-10-0
{{< /text >}}

Now, the situation is as below:

{{< image width="70%"
    link="/docs/setup/upgrade/canary/tags-updated.png"
    caption="Namespace labels unchanged but now all namespaces pointed to 1-10-0"
>}}

Restarting injected workloads in the namespaces marked `prod-stable` will now result in those workloads using the `1-10-0`
control plane. Notice that no namespace relabeling was required to migrate workloads to the new revision.

### Default tag

The revision pointed to by the tag `default` is considered the ***default revision*** and has additional semantic meaning.

> The `default` tag will inject for the `istio-injection=enabled` namespace selector and `sidecar.istio.io/inject=true` object

Reviewer: ```suggestion
The default revision will inject sidecars for the `istio-injection=enabled` namespace selector and `sidecar.istio.io/inject=true` object
```

(The _tag_ doesn't actively inject.)

Author: Fixed.

selector in addition to the `istio.io/rev=default` selectors. This makes it possible to migrate from using non-revisioned Istio to using

> a revision **entirely without relabeling namespaces**. To make a revision `1-10-0` the default, run:

Reviewer: ```suggestion
a revision without relabeling namespaces. To make a revision `1-10-0` the default, run:
```
